src/java/org/apache/cassandra/config/RetrySpec.java

```diff
@@ -0,0 +1,182 @@
+            return Integer.toString(value);
+        }
+    }
+
+    public static class Partial extends RetrySpec
+    {
+        public Partial()
+        {
+            this.type = null;
+            this.maxAttempts = null;
+            this.baseSleepTime = null;
+            this.maxSleepTime = null;
+        }
+
+        public RetrySpec withDefaults(RetrySpec defaultValues)
+        {
+            Type type = nonNull(this.type, defaultValues.getType(), DEFAULT_TYPE);
+            MaxAttempt maxAttempts = nonNull(this.maxAttempts, defaultValues.getMaxAttempts(), DEFAULT_MAX_ATTEMPTS);
+            LongNanosecondsBound baseSleepTime = nonNull(this.baseSleepTime, defaultValues.getBaseSleepTime(), DEFAULT_BASE_SLEEP);
+            LongNanosecondsBound maxSleepTime = nonNull(this.maxSleepTime, defaultValues.getMaxSleepTime(), DEFAULT_MAX_SLEEP);
+            return new RetrySpec(type, maxAttempts, baseSleepTime, maxSleepTime);
+        }
+
+        private static <T> T nonNull(@Nullable T left, @Nullable T right, T defaultValue)
+        {
+            if (left != null)
+                return left;
+            if (right != null)
+                return right;
+            return defaultValue;
+        }
+    }
+
+    private static final Type DEFAULT_TYPE = Type.Expoential;
+    private static final MaxAttempt DEFAULT_MAX_ATTEMPTS = MaxAttempt.DISABLED;
+    private static final LongNanosecondsBound DEFAULT_BASE_SLEEP = new LongNanosecondsBound("200ms");
+    private static final LongNanosecondsBound DEFAULT_MAX_SLEEP = new LongNanosecondsBound("1s");
+
+    public Type type = DEFAULT_TYPE;
+    /**
+     * Represents how many retry attempts are allowed.  If the value is 2, this will cause 2 retries + 1 original request, for a total of 3 requests!
+     * <p/>
+     * To disable, set to 0.
+     * <p/>
+     * To make it unbound, used -1
+     */
+    public MaxAttempt maxAttempts = DEFAULT_MAX_ATTEMPTS; // 2 retries, 1 original request; so 3 total
+    // TODO (api): These are really only useful if type=Expoential, but we don't have dynamic deserialization so can not really split into a type specific parser atm...
+    public LongNanosecondsBound baseSleepTime = DEFAULT_BASE_SLEEP;
+    public LongNanosecondsBound maxSleepTime = DEFAULT_MAX_SLEEP;
+
+    public RetrySpec()
+    {
+    }
+
+    public RetrySpec(Type type, MaxAttempt maxAttempts, LongNanosecondsBound baseSleepTime, LongNanosecondsBound maxSleepTime)
+    {
+        this.type = type;
+        this.maxAttempts = maxAttempts;
+        this.baseSleepTime = baseSleepTime;
+        this.maxSleepTime = maxSleepTime;
+    }
+
+    public boolean isEnabled()
+    {
+        return maxAttempts != MaxAttempt.DISABLED;
+    }
+
+    public void setEnabled(boolean enabled)
+    {
+        if (!enabled)
+        {
+            maxAttempts = MaxAttempt.DISABLED;
+        }
+        else if (maxAttempts == MaxAttempt.DISABLED)
+        {
+            maxAttempts = new MaxAttempt(2);
```

Comment:
nit: Arbitrarily set to 2 retries if we're enabling?

Reply:
what do you recommend?

Reply:
I think I'm just wary of being able to set this enabled at all without explicitly setting `max_attempts`. If we're going to do it though, I guess 3 total attempts is reasonable.